### DISCUSSION TITLE
Help with MIDI input, cant get it working

### DISCUSSION THREAD

**Discussion by kemikal** [tttapa/Control-Surface#613] [Q&A]
Hi everyone, thanks for a great library!
Ive used it to create a custom sampler/synth for a art exhibition were visitors can remix the music and the artwork by playing with the synthesizer. So I got all the midi output working as it should. 
(heres a short clip to demonstrate what I am doing;
https://www.youtube.com/watch?v=kDXZwpkwMok)

But now I would need to send a midi out signal from the computer at certain intervals to "reset" all the lights in the synth.
I am using the Metro M4 Grand Central. And I can see on the card that its flashing when I send the signal (MIDI out from computer).

But I cant get the midi input examples to work. 
When just pasting the simple starter examples from:
https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/df7/midi-tutorial.html

I cant get it working. Trying the "Customizable callback" I get error from "override" (Marked Override, but does not override) removing the keyword it builds but I see no output to serial monitor.

Trying the simple example I get error (expected template-name before '<' token)
on line;
struct MyMIDI_Callbacks : FineGrainedMIDI_Callbacks<MyMIDI_Callbacks> {

I think I should get It working if I just understand how to react to a incoming midi message!

Appreciate any help!
/ j

**Comment by MiOiM** [comment]
Hey @kemikal!

Did you manage to solve the problem with the override error when compiling the midi input examples? I am having the same issue, just posted a question here...

**Comment by kemikal** [comment]
Hi MiOiM!
Nope, have not got midi input from computer to control surface working. Everything else is working nice, was just not able to figure this out.

**Reply by MiOiM** [reply]
Oh, ok...
By any chance are you using VSC and PlatformIO? 
I did run a quick test with the Arduino IDE and it compiled correctly. I believe the problem is with VSC or PlatformIO...

**Comment by tttapa** [comment]
It sounds like you're using a new tutorial with an older version of the library. Try installing the latest `master` version from GitHub, or follow the documentation for the specific version you have installed, see https://tttapa.github.io/Control-Surface-doc/.

IIRC, you can install the latest version using 
```
pio lib install https://github.com/tttapa/Control-Surface.git#master
```
